common.go

```diff
@@ -1,71 +1,171 @@
 package slinguist
 
+import (
+	"path/filepath"
+	"strings"
+)
+
+// OtherLanguage is used as a zero value when a function can not return a specific language.
 const OtherLanguage = "Other"
 
-var (
-	ExtensionsByLanguage map[string][]string
-	ignoredExtensions    = map[string]bool{
-		".asc": true, ".cgi": true, ".fcgi": true, ".gml": true, ".fx": true,
-		".vhost": true,
+var strategies = []func(filename string, content []byte) (languages []string){
+	getLanguagesByModeline,
```

Comment:
why we made the strategies private? I prefer having `strategies` public and also the functions allowing to the user choose his own strategies if they want.